scipy/optimize/_minpack_py.py

```diff
@@ -639,6 +641,19 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         `mesg`, and `ier`.
 
         .. versionadded:: 1.9
+    nan_policy : {'propagate', 'raise', 'omit', None}, optional
+        Defines how to handle when input contains nan.
+        The following options are available (default is 'None'):
+
+          * 'propagate': just propagate nan values
+          * 'raise': throws an error
+          * 'omit': performs the calculations ignoring nan values
+          * None: check nan values based on `check_finite` value.
```

Comment:
This is what `None` will mean in stats, when it is introduced.

> no special handling of NaNs is performed; the behavior when NaNs are present is implementation-dependent and may change.

Here in curve fit, I think that would just be qualified:

> no special handling of NaNs is performed (except what is done by `check_finite`); the behavior when NaNs are present is implementation-dependent and may change.